spec/models/user_spec.rb

```diff
@@ -825,6 +836,19 @@
         user.suspended_at = Time.zone.now
         user.reinstated_at = nil
       end
+
+      it 'destroys SuspendedEmail records for each email address' do
+        suspended_email = double
+        expect(SuspendedEmail).to receive(:generate_email_digest).
+          with(user.email_addresses.last.email).
+          and_return('digest1')
+        expect(SuspendedEmail).to receive(:find_by).with(
+          digested_base_email: 'digest1',
+        ).and_return(suspended_email)
+        expect(suspended_email).to receive(:destroy)
+        user.reinstate!
```

Comment:
ditto, I think we should check for the real side effect intead of stub

```suggestion
        expect { user.reinstate! }.to(change { SupendedEmail.find_with_email('...') }.to(nil))
```